packages/react-server-webpack-plugin/README.md

```diff
@@ -88,6 +88,52 @@ started react-server on localhost:PORT
 
 If you point your browser at `localhost:PORT` you should see "I am an app". :)
 
+### Automatic Props
+
+The plugin passes some props to your application automatically.
+
+```typescript
+interface DefaultProps {
+  /*
+   The full WHATWG URL object for the initial request
+  */
+  url: URL;
+
+  /*
+   An object containing any data sent via the `x-quilt-data` header.
+   This header is used by `quilt_rails` for sending data directly from ruby to React.
+  */
+  data: Record<string, any>;
+}
+```
+
+These props are provided during both server-side and client-side rendering, so they can be used without any special logic around what environment your app is in.
+
+These props can be used to feed initial data into your application for use with libraries such as `react-router`, as well as for simple app logic.
+
+```tsx
+// index.jsx
+import React from 'react';
+
+export default function App({url, data}: {url: Url, data: Record<string, any>}) {
+  const {href} = url;
+  if (href.endsWith('/about')) {
+    return <div>this is an about page</div>;
+  }
+
+  return (
```

Comment:
running into a small integration issue with react-router
```
import {Router} from '@shopify/react-router';
<Router location={url}>
```
👆 this fail type check because `location` is typed `location?: string;`

we can use `url.href` or change the type to `location?: string | URL;` and everything would still work

Reply:
url.href is the right choice, that's what I did in the branch I linked

Reply:
Yes, I understand that I can use `url.href`. The point I was trying to make it we don't have to.

Our `Router` only use `location` to pass to `react-router`'s `StaticRouter`

Their doc show example of the following

```
<StaticRouter location={req.url} context={context}>
        <App />
</StaticRouter>
```

which mean with a type change `Router` can accept object as well as string.

Reply:
ahhh, sorry I misunderstood your comment. Seems like it makes sense to update the type indeed then!